sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -238,16 +238,29 @@ public void startBundle(StartBundleContext context) {
 
     @ProcessElement
     public void processElement(ProcessContext context) {
-      Object[] avaticaRow = rowToAvatica(context.element());
-      if (avaticaRow.length == 1) {
-        values.add(avaticaRow[0]);
+      values.add(rowToObjectArray(context.element()));
+    }
+  }
+
+  private static List<Object> objectArrayToAvatica(Queue<Object[]> values) {
+    List<Object> ret = new ArrayList<>();
+    Iterator<Object[]> iterator = values.iterator();
+    while (iterator.hasNext()) {
+      Object[] objects = iterator.next();
```

Comment:
nit: This line and the two above are the same as `for (Object[] objects : values) {`. I would suspect there is a way to simplify this more with `stream`, might ask @akedin.

Reply:
I use `stream` API now. I misunderstood `stream` API and thought that would bring in concurrent issue. Now I figure it out that `stream` API returns a sequential stream.